### DISCUSSION TITLE
Getting Domain in Quartz with web app

### DISCUSSION THREAD

**Discussion by eminyuce** [quartznet/quartznet#300] [General] [answered]
Hi, I am using quartz in the asp.net mvc app and trying to find out domain name from IJob implementation. Is it possible to get domain name inside IJob implementation?

I am using Ninject to initiate  scheduler jobs.

```
    kernel.Bind<ISchedulerFactory>().To<NinjectSchedulerFactory>();
        kernel.Bind<IScheduler>().ToMethod(ctx => ctx.Kernel.Get<ISchedulerFactory>().GetScheduler()).InSingletonScope();
        kernel.Bind<IBaseTasksScheduler>().To<LiquidTasksScheduler>().WithConstructorArgument("scheduler", kernel.Get<IScheduler>());
        kernel.Get<IBaseTasksScheduler>().Start(); 
```

 public class TestJob : IJob
    {
        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();

```
    public void Execute(IJobExecutionContext context)
    {
      var domain= GetWebAppDomainNameFromSomewhere();
        Logger.Trace("TestJob is ScheduledTasks running.");
    }
}
```

**Comment by vgl** [comment, marked as answer]
Hi,
I think you can get it at runtime using several ways.
1 call web app method of asp.net mvc app in execute method using http client, or whatever rest library you like. But if you need to know domain name before making app call then this variant is not for you.
2 put domain name into JobDataMap  when job is creating and in execute method retrieve it from that dictionary.
